src/domain/settings/taxes/selectable-table.tsx

```diff
@@ -0,0 +1,227 @@
+      autoResetSelectedRows: false,
+      autoResetPage: false,
+      getRowId: (row) => row.id,
+    },
+    usePagination,
+    useRowSelect,
+    (hooks) => {
+      hooks.visibleColumns.push((columns) => [
+        // Let's make a column for selection
+        {
+          id: "selection",
+          // The header can use the table's getToggleAllRowsSelectedProps method
+          // to render a checkbox
+          Header: ({ getToggleAllRowsSelectedProps }) => {
+            return (
+              <div>
+                <IndeterminateCheckbox {...getToggleAllRowsSelectedProps()} />
+              </div>
+            )
+          },
+          // The cell can use the individual row's getToggleRowSelectedProps method
+          // to the render a checkbox
+          Cell: ({ row }) => {
+            return (
+              <div>
+                <IndeterminateCheckbox {...row.getToggleRowSelectedProps()} />
+              </div>
+            )
+          },
+        },
+        ...columns,
+      ])
+    }
+  )
+
+  useEffect(() => {
+    onChange(Object.keys(selectedRowIds))
+  }, [selectedRowIds])
+
+  const handleNext = () => {
+    if (canNextPage) {
+      handleQueryChange({
+        ...pagination,
+        offset: pagination.offset + pagination.limit,
+      })
+      nextPage()
+    }
+  }
+
+  const handlePrev = () => {
+    if (canPreviousPage) {
+      handleQueryChange({
+        ...pagination,
+        offset: Math.max(pagination.offset - pagination.limit, 0),
+      })
+      previousPage()
+    }
+  }
+
+  return (
+    <div>
+      <div className="inter-base-semibold my-large">{label}</div>
+      <Table
+        immediateSearchFocus={showSearch}
+        enableSearch={showSearch}
+        searchPlaceholder="Search Products.."
+        handleSearch={onSearch}
+        {...getTableProps()}
+      >
+        <Table.Body {...getTableBodyProps()}>
+          {isLoading ? (
+            <Spinner size="large" />
+          ) : (
+            rows.map((row, i) => {
+              prepareRow(row)
+              return (
+                <Table.Row {...row.getRowProps()}>
+                  {row.cells.map((cell) => {
+                    return (
+                      <Table.Cell {...cell.getCellProps()}>
+                        {cell.render("Cell")}
+                      </Table.Cell>
+                    )
+                  })}
+                </Table.Row>
+              )
+            })
+          )}
+        </Table.Body>
+      </Table>
+      <TablePagination
+        count={totalCount!}
+        limit={pagination.limit}
+        offset={pagination.offset}
+        pageSize={pagination.offset + rows.length}
+        title={objectName}
+        currentPage={pageIndex + 1}
+        pageCount={pageCount}
+        nextPage={handleNext}
+        prevPage={handlePrev}
+        hasNext={canNextPage}
+        hasPrev={canPreviousPage}
+      />
+    </div>
+  )
+}
+
+const IndeterminateCheckbox = React.forwardRef(
```

Comment:
Should we replace this with our molecule [`indeterminate-checkbox`](https://github.com/medusajs/admin/blob/master/src/components/molecules/indeterminate-checkbox/index.tsx) to avoid having duplicate code?